### ISSUE TITLE
[Feature Request]: Allow binding keys to diagonal movement

### ISSUE THREAD

**Issue by sidav** [diasurgical/devilutionX#7607]
### Feature Type

Gamepad Controls

### Describe

On devices like Miyoo Mini, where there are no diagonals on the D-Pad, the gameplay is not that comfortable because it's not possible to bind e.g. "d-pad up" to "move up and right". Example of where it's needed is the bridge that leads to Adria - I just press the "D-pad down" and "D-pad right" with two fingers.
It would be great to have separate bind options for diagonal movement.

**Comment by AJenbo** [comment]
What would you bind it to? How would you move to the right if the right button is mapped to south east?

It works perfectly fine on RG300 which has a similar d-pad:

![image](https://github.com/user-attachments/assets/fd77112c-684c-42f4-8459-5d6d86a7d748)

If you cannot push the d-pad in a diagonal direction with on finger there will be a lot of games designed for dpad that won't be very playable.

- Contra series
- Sonic Advance
- Metal Gear 
- Street Fighter
- Tetris Attack

**Comment by sidav** [comment]
I'd simply bind it to the D-Pad and unbind the non-diagonals. 

> How would you move to the right if the right button is mapped to south east?
Via moving south-east and then north-east. I don't remember any places in the game where moving right is strictly neccessary, whereas there are tons of them where moving diagonally is.

**Comment by Trihedraf** [comment]
What are you talking about? Diagonals work just fine on miyoo mini. https://github.com/user-attachments/assets/54ec7826-d48c-4ce2-9568-07fcab5512c3

What version of DevilutionX are your running and where did you get it?

**Comment by AJenbo** [comment]
I guess your assumption is that it would be implemented in a way where pushing two diagonals would result in a non diagonal.

**Comment by sidav** [comment]
> I guess your assumption is that it would be implemented in a way where pushing two diagonals would result in a non diagonal.

Why? My assumption is that it's possible to play without the non-diagonal movement altogether, as there are no places in the whole game which you can't reach without those "cardinal directions", so you actually wouldn't _need_ pressing two keys simultaneously anywhere. From the other hand, there are tons of places where you can't reach with those cardinals only.

> What are you talking about? Diagonals work just fine on miyoo mini. https://github.com/user-attachments/assets/54ec7826-d48c-4ce2-9568-07fcab5512c3
> What version of DevilutionX are your running and where did you get it?

It seems it's not a problem with any version, it seems more like a problem with my fingers. Well, I thought it wouldn't be too complicated to add and it's not game breaking  in any way (as this movement already is in the game, it's just not directly accessible). Thanks for the attention anyways.

**Comment by AJenbo** [comment]
Hm it's possible that would work. I'll leave the issue open for now.

**Comment by julealgon** [comment]
@sidav so you are basically asking for that toggle that exists in the PS1 version of D1 to use up/down/left/right for cardinal movement instead of "absolute screen" movement, correct?

**Comment by StephenCWills** [comment]
The first thing that came to my mind when I read this issue was Final Fantasy Tactics Advance, which offers "Type A" and "Type B" control schemes. The game is an isometric turn-based strategy. One of the control schemes uses d-pad up to move the cursor NE, the other uses d-pad up to move the cursor NW (and the other d-pad directions are relative, of course).

I think regardless of what sidav meant, I'm not against implementing these "Type A" and "Type B" control schemes for DevilutionX. At the very least, he has a point that it should make moving through doorways/archways easier.

**Comment by sidav** [comment]
> @sidav so you are basically asking for that toggle that exists in the PS1 version of D1 to use up/down/left/right for cardinal movement instead of "absolute screen" movement, correct?
Yeah, either that, or separate optional keybinds for diagonals. Anyway it would be more comfortable than it is right now.
